src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -31,21 +33,35 @@ class ServicePrincipalMigrationInfo:
 
 @dataclass
 class StorageCredentialValidationResult:
-    name: str
+    name: str | None
     application_id: str
-    read_only: bool
+    read_only: bool | None
     validated_on: str
-    directory_id: str | None = None
+    directory_id: str | None = None  # str when storage credential created for AzureServicePrincipal
     failures: list[str] | None = None
 
     @classmethod
-    def from_validation(cls, permission_mapping: StoragePermissionMapping, failures: list[str] | None):
+    def from_storage_credential_info(
+        cls, storage_credential_info: StorageCredentialInfo, validated_on: str, failures: list[str] | None
```

Comment:
```suggestion
        cls, storage_credential_info: StorageCredentialInfo, validated_on: str, failures: list[str] | None,
```
nit: if you add trailing comma, all arguments will be put on a separate line by `ruff`